Helm_charts (#182)
percona/everest-doc: docs/uninstall/uninstall_everest_helm.md

# Uninstall Percona Everest using Helm

If you installed Everest using Helm, follow the instructions below to uninstall.

## Uninstall the database namespaces

If you have created any database namespaces other than the default namespace, ensure to delete them first.

```sh

> helm uninstall everest -n <your_db_namespace>
> kubectl delete ns <your_db_namespace>

Reviewer: This style is more in line with the rest of the docs
```suggestion
helm uninstall everest -n <DBNamespace>
kubectl delete ns <DBNamespace>
```

```

!!! note
    This runs a chart hook that cleans up your database resources. Although it is not recommended, you can skip this step by specifying `cleanupOnUninstall=false`.

## Uninstall Percona Everest

To uninstall Percona Everest, follow these steps:
{.power-number}

1. Run the following command:

    ```sh
    helm uninstall everest-core -n everest-system
    kubectl delete ns everest-system
    ```

2. Remove the unused [Custom Resource Definitions (CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).

    !!! caution alert alert-warning "warning"

        Deleting CRDs can potentially cause issues with any custom resources that depend on those definitions within the kubernetes cluster. Ensure that you do not remove any CRDs that are being used by operator deployments outside of Everest.

> ## Remove CRDs

Reviewer: This section is out of date, please find the up-to-date list of CRDs in the main branch.

Author: Fixed.

# Uninstall Percona Everest using Helm

> If you are looking for an alternative method to uninstall Percona Everest, you can also consider using the Helm charts.

Reviewer 2: ```suggestion
If you installed Everest using Helm, follow the instructions below to uninstall.
```

    customresourcedefinition.apiextensions.k8s.io "perconaservermongodbbackups.psmdb.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io "perconaservermongodbrestores.psmdb.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io "perconaservermongodbs.psmdb.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io "perconapgbackups.pgv2.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io "perconapgrestores.pgv2.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io "perconapgclusters.pgv2.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io "postgresclusters.postgres-operator.crunchydata.com" deleted
    ```

#### Remove CRDs of Percona's PostgreSQL operator

If you are using the PostgreSQL operator from **Crunchy Data**, do not remove the `postgresclusters.postgres-operator.crunchydata.com` CRD. 

In this case, run the following command:

```sh
cat <<EOF | xargs kubectl delete crd
perconapgbackups.pgv2.percona.com
perconapgrestores.pgv2.percona.com
perconapgclusters.pgv2.percona.com
EOF
```

> 
> 
> 
> 
> 
> 
> 

Reviewer 2: ```suggestion
```

# Uninstall Percona Everest using Helm

If you are looking for an alternative method to uninstall Percona Everest, you can also consider using the Helm charts.

## Uninstall the database namespaces

If you have created any database namespaces other than the default namespace, ensure to delete them first.

```sh
helm uninstall everest -n <your_db_namespace>
kubectl delete ns <your_db_namespace>
```

!!! note
    This runs a chart hook that cleans up your database resources. Although it is not recommended, you can skip this step by specifying `cleanupOnUninstall=false`.

## Uninstall Percona Everest

To uninstall Percona Everest, run the following command:

```sh
helm uninstall everest-core -n everest-system
kubectl delete ns everest-system
```

> 

Reviewer: We need the https://everest-release-branch-pr-182.onrender.com/uninstall/uninstallEverest.html#how-to-remove-crds section

Author: Added.

    "perconaxtradbclusters.pxc.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io 
    "perconaservermongodbbackups.psmdb.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io 
    "perconaservermongodbrestores.psmdb.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io 
    "perconaservermongodbs.psmdb.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io 
    "perconapgbackups.pgv2.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io 
    "perconapgclusters.pgv2.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io 
    "perconapgrestores.pgv2.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io 
    "perconapgupgrades.pgv2.percona.com" deleted
    customresourcedefinition.apiextensions.k8s.io 
    "crunchybridgeclusters.postgres-operator.crunchydata.com" deleted
    customresourcedefinition.apiextensions.k8s.io "pgadmins.postgres- 
    operator.crunchydata.com" deleted
    customresourcedefinition.apiextensions.k8s.io "pgupgrades.postgres- 
    operator.crunchydata.com" deleted
    customresourcedefinition.apiextensions.k8s.io 

>     "postgresclusters.postgres-operator.crunchydata.com" deleted

Reviewer: Why did you add line breaks for this output? Now, it doesn't reflect what the user sees in the terminal when running the command...

Author: I attempted to remove the copy button, which caused this issue. I have undone the changes.
